daemon/logger/awslogs/cloudwatchlogs_test.go

```diff
@@ -471,6 +489,186 @@ func TestCollectBatchTicker(t *testing.T) {
 
 }
 
+func TestCollectBatchMultilinePattern(t *testing.T) {
+	mockClient := newMockClient()
+	multilinePattern := regexp.MustCompile("xxxx")
+	stream := &logStream{
+		client:           mockClient,
+		logGroupName:     groupName,
+		logStreamName:    streamName,
+		multilinePattern: multilinePattern,
+		sequenceToken:    aws.String(sequenceToken),
+		messages:         make(chan *logger.Message),
+	}
+	mockClient.putLogEventsResult <- &putLogEventsResult{
+		successResult: &cloudwatchlogs.PutLogEventsOutput{
+			NextSequenceToken: aws.String(nextSequenceToken),
+		},
+	}
+	ticks := make(chan time.Time)
+	newTicker = func(_ time.Duration) *time.Ticker {
+		return &time.Ticker{
+			C: ticks,
+		}
+	}
+
+	go stream.collectBatch()
+
+	stream.Log(&logger.Message{
+		Line:      []byte(logline),
+		Timestamp: time.Now(),
+	})
+	stream.Log(&logger.Message{
+		Line:      []byte(logline),
+		Timestamp: time.Now(),
+	})
+	stream.Log(&logger.Message{
+		Line:      []byte("xxxx " + logline),
+		Timestamp: time.Now(),
+	})
+
+	ticks <- time.Time{}
+
+	// Verify single multiline event
+	argument := <-mockClient.putLogEventsArgument
+	if argument == nil {
```

Comment:
Since the time when I originally wrote these tests and now, there's a new assertion library that has been added.  I'd prefer if the new tests you write use the assertion library instead of following the existing style in this file (and I'd be happy to follow up and change the rest of the tests).

Import `github.com/stretchr/testify/assert` (which is already vendored) and then you can do things like:

```go
assert.NotNil(t, argument, "Expected non-nil PutLogEventsInput")
assert.Equal(t, 1, len(argument.LogEvents), "Unexpected LogEvents length")
assert.Equal(t, logline+logline, *argument.LogEvents[0].Message, "Unexpected message")
```